### PR TITLE
Backport 'Initialize and update the ydoc path property' #342

### LINKED ISSUE #341: Populating the path property of the ydoc
## Description

The path property of the ydoc is not populated.

This is probably a regression included in https://github.com/jupyterlab/jupyter-collaboration/pull/174.
The path used to be set here: https://github.com/jupyterlab/jupyter-collaboration/blob/547667399adeefcf2729a2c76b4526e752efdbf6/jupyter_server_ydoc/ydoc.py#L127

## Reproduce

Open a document in collaborative mode and try to get the path from the ydoc.

## Expected behavior

The path to be populated.

## Suggestion

Set the path should be trivial at initialization, i.e. at https://github.com/jupyterlab/jupyter-collaboration/blob/b8fafc0646e03ce2dcbd27b71267b1b99c976547/projects/jupyter-server-ydoc/jupyter_server_ydoc/rooms.py#L44

Then we should probably update the path if it has changed, probably from https://github.com/jupyterlab/jupyter-collaboration/blob/b8fafc0646e03ce2dcbd27b71267b1b99c976547/projects/jupyter-server-ydoc/jupyter_server_ydoc/loaders.py#L183

### PR DESCRIPTION
**Pull request by brichet**
This PR is a backport of #342 to fix #341 on 2.x branch

### CODE REVIEW

**Comment by davidbrochart** [comment]
@brichet did we forget to merge this one?

**Comment by brichet** [comment]
Probably, but I don't know why the test are failing...
I'll rebase and we'll see

**Comment by brichet** [comment]
> Probably, but I don't know why the test are failing...

Actually these tests seems to fail with every PR on `2.x` branch.

File: jupyter_collaboration/loaders.py [hunk 1]
```diff
@@ -97,6 +106,8 @@ def unobserve(self, id: str) -> None:
                 id (str): Room ID
         """
         del self._subscriptions[id]
+        if id in self._filepath_subscriptions.keys():
```
Comment (davidbrochart) [inline jupyter_collaboration/loaders.py:109]:
```suggestion
        if id in self._filepath_subscriptions:
```

**Review by davidbrochart — APPROVED** [review]

**Pull request merged**